### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by cntkillme**
* Add English translations of research notes.
* Remove crash notice from `StrMap*::erase(str)` as it does not crash on Windows, Imperivm III v1.40 (maybe it only crashes under Wine for some reason?).
* Document `IsFlagSet(int, int), SetFlag(int*, int, bool)`
* Document Item members (`Item::use_count, Item::custom_data, Item::SetUseCount, Item::SetCustomData, Item::Use`)
* Document Item methods (`Squad::TakeNearbyItems, ItemHolder::AddItem, ItemHolder::RemoveItem, ItemHolder::HasItem, Obj::HasItem, Obj::UseItem`)
* Document Item free functions (`GetThis, GetUseCount, SetUseCount, ItemUsed, GetItem`)
* Document many selection functions
    - `sel, selo, selu, selb, sels, selsq, selg, selh`
    - `SelHealth, SelMaxHealth, SelAvgLevel, SelAvgArmor, SelAvgDamage, SelAvgFood, SelAvgStamina`
    - `SelectionGold, SelectionFood`
    - `SelSquad, SelSquadLeader`
* Document diplomacy functions (`ClearDiplomacy, DiplAreAllied, DiplCeaseFire, DiplGetCeaseFire, DiplShareView, DiplGetShareView, DiplShareSupport, DiplGetShareSupport, DiplShareControl, DiplGetShareControl`)
* Document View functions (`IsViewLocked, LockView, UnlockView, StartViewFollow, StopViewFollow, View, ViewSlot`)
* Document some Map functions (`GetMapRect, MapName`)
* Document zoom map functions (`BlockMiniMap, _ZoomMapLastShownTime, ShowZoomMap, HideZoomMap, ToggleZoomMap`)

New documentation is only in English, a `description` field still needs to be added.

### CODE REVIEW

File: dist/display.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ function renderNotForSequences(func) {
 
 function renderResearchNotes(func) {
     if (func.research_needed) {
```
Comment (david-campos) [inline dist/display.js]:
```suggestion
    if (func["research_needed_" + lang] || func.research_needed) {
```
so it takes in account if the translation for english exists even if no spanish one is provided

File: dist/data.js [hunk 2]
```diff
@@ -7240,7 +7262,9 @@ const THE_OBJ = (function () {
             "of": null,
             "of_ptr": false,
             "type": "method",
-            "params": []
+            "params": [],
+            "description_en": "Hides the map",
```
Comment (david-campos) [inline dist/data.js]: Missing final stop "."

File: dist/data.js [hunk 3]
```diff
@@ -10721,7 +10826,8 @@ const THE_OBJ = (function () {
             "of": null,
             "of_ptr": false,
             "type": "method",
-            "params": []
+            "params": [],
+            "description_en": "Returns the current item. This should only be used in an item script, otherwise an invalid Item is returned."
```
Comment (david-campos) [inline dist/data.js]: Item On Die seems to return an invalid Item as well, I think this should be noted

File: dist/data.js [hunk 4]
```diff
@@ -12513,7 +12619,9 @@ const THE_OBJ = (function () {
             "of": null,
             "of_ptr": false,
             "type": "method",
-            "params": []
+            "params": [],
+            "description_en": "Similar to <a href='#ViewPos'>ViewPos</a>, but scaled down in some way. More research is needed, the x component is seemed to be divided by 128 but not the y component?",
```
Comment (david-campos) [inline dist/data.js]: Turn the last sentence into a research note :)

File: dist/data.js [hunk 5]
```diff
@@ -16594,7 +16731,9 @@ const THE_OBJ = (function () {
             "of": null,
             "of_ptr": false,
             "type": "method",
-            "params": [{"name": "number", "type": 1, "is_ptr": false}, {"name": "number", "type": 1, "is_ptr": false}]
+            "params": [{"name": "player1", "type": 1, "is_ptr": false}, {"name": "player2", "type": 1, "is_ptr": false}],
+            "description_en": "Returns true if <tt>player1</tt> and <tt>player2</tt> cannot attack each other, or false otherwise.",
```
Comment (david-campos) [inline dist/data.js]: Have you checked this? I run with wine so there might be something wrong but I get false even when ceasing fire between the players. In my case, I need to have all shared: view, control, supply, cease fire and allied victory. Please review this. Also detail how _not-symmetrical_ relations affect it (some methods behave weird with these).

**Review by david-campos — CHANGES_REQUESTED** [review]
- [ ] Check the comments in the code and correct the marked points.
- [ ] The names for all the parameters need to be moved from "name" to "name_en" as they are the english ones.
- [ ] The methods which are overloads of other methods are easy to miss since the spanish description is copied but without the "overload" indication. I think the best solution is to remove the text about overloading from the english translation and adding a field `"overrides": "id of the overrided method"`. Although this is not exactly overriding, as it seems they do the same. We should talk about this in Discord when you have time.
- [ ] The ones which are useful only inside Item scripts need to be marked, so please add a field to these ones like this: `"itemScriptOnly": true`. Don't need to make anything with the field, I will add later an icon on the top similar to the ones which are not useful in sequences or something similar.

Great job! Lots of methods documented! Thank you very much, really. Correct those minor things and I will merge it. :)

File: dist/data.js (hunk 5 above)
Comment (cntkillme) [reply dist/data.js]: I only tested this with allied victory enabled. After doing more testing, it seems like both players must have Allied victory set and symmetrical cease fire for it to return true. Will update to account for this.

For future reference: the functions `_PlayersAlly` sets the allied victory flag between two players and `_PlayersMakeEnemies` clears the flag. Will eventually document these once I finish fixing all the current issues.

Comment (cntkillme) [reply dist/data.js]: Also the Allied victory flag is different from the other flags in that it seems to be symmetric (`_PlayersAlly(1, 2) <=> _PlayersAlly(2, 1)`)

**Pull request merged**